src/processor.cpp

```diff
@@ -0,0 +1,401 @@
+    }
+}
+
+void process_data::load( JsonObject &jo, const std::string &src )
+{
+    processes_data.load( jo, src );
+}
+
+void process_data::check()
+{
+    processes_data.check();
+}
+
+void process_data::reset()
+{
+    processes_data.reset();
+}
+
+void interact_with_processor( const tripoint &examp, player &p )
+{
+    //Get processor id at examp
+    processor_id pid = get_processor_id( examp );
+    //Do I need the debugmsg-es here or do they get checked in process_data::load?
+    if( !processors_data.is_valid( pid ) ) {
+        debugmsg( "Examined %s has action interact_with_processor, but has no processor data associated", g->m.furn( examp ).id().c_str() );
+        return;
+    }
+    //Get data associated with said processor id
+    const processor_data &processor = processors_data.obj( pid );
+    std::vector<auto> possible_processes;
+    for( auto &p : processor.processes() ) {
+        if( p.is_valid() ){
+            possible_processes.push_back( process_data.obj( process_id( p ) ) );
+        } else {
+            //Debugmsg probably not exactly right,
+            //I'm trying to return the processor name and the problematic process name too
+            debugmsg( "Examined %s has non-valid process data %s2 associated", processor.id().c_str(), get_process_id( p ).c_str() );
+            return;
+        }
+    }
+
+    //  Figure out which process to run, easier to work with than
+    //  trying to figure out desired process from items on tile,
+    //  Can become a mess with processor with multiple possible processes
+    std::vector<std::string> p_names;
+    for( auto p : possible_processes ){
+        p_names.push_back( p.name );
+    }
+    int p_index = 0;
+    if( p_names.size() > 1 ){
+        p_names.push_back(_("Cancel"));
+        p_index = menu_vec(true, _("Possible processes"), p_names);
+        if (p_index == (int)p_names.size() - 1) {
+            return;
+        }
+    }
+
+    auto active_process = possible_processes[p_index];
+
+    //Collect possible processed item id-s and fuels into a vector
+    //Use said vector to find out if we have usable items on the tile
+    //TODO: Handle processes with multiple possible inputs, see kiln
+    //TODO: Handle processes requiring multiple input materials
+    std::vector<itype_id> processed_materials;
+    for( auto &i : active_process.components ) {
+            //Not sure about this line
+            //I want to collect the possible components item id's into the vector
+            processed_materials.push_back( itype_id( i ) );
+    }
+    if( processed_materials.empty() ){
+        debugmsg( "Examined %s has no reagents associated", active_process.id().c_str() );
+        return;
+    }
+
+    //You did say one function to handle all
+    //processors not in the middle of a process
+    //Basically, this furniture type denotes the type
+    //where it is not in the middle of making something
+    if(!processor.is_processing){
```

Comment:
This will be a method, not a value. That is, `processor.is_processing( examp )` or something like that.
That is because you can't cache anything in the processor TYPE - this one has to stay immutable.

But a better way to do that than having a `processor.is_processing` method is having two `iexamine` methods:
* Start processing
* Is processing (also includes end processing)